components/ProductCard.js

```diff
@@ -0,0 +1,117 @@
+import styled from 'styled-components';
+import QuantitySelector from './QuantitySelector';
+import { useState } from 'react';
+
+export default function ProductCard({ product }) {
+  const [showDetails, setShowDetails] = useState();
+
+  return (
+    <>
+      <StyledCard key={product.id}>
+        <StyledImage src={product.image} alt={product.name} />
+        <MainInfoWrapper>
+          <StyledProductName>{product.name}</StyledProductName>
+          <StyledWSPrice>WS {product.WSprice.toPrecision(4)} €</StyledWSPrice>
+          <MainInfoWrapperRow>
+            <StyledMoreInfoButton
+              type="button"
+              key={product.id}
+              onClick={() => {
+                setShowDetails(!showDetails);
+              }}
+            >
+              {showDetails ? 'Less Info' : 'More Info'}
+            </StyledMoreInfoButton>
+            <QuantitySelector product={product} />
+          </MainInfoWrapperRow>
+        </MainInfoWrapper>
+      </StyledCard>
+      {showDetails ? (
+        <ExtraInfoWrapper>
+          <StyledExtraInfo>
+            <strong>RRP: </strong> {product.RRPprice.toPrecision(4)} €
+            <br /> <strong>BARCODE:</strong> {product.id}
+          </StyledExtraInfo>
+          <StyledExtraInfo>{product.description}</StyledExtraInfo>
+        </ExtraInfoWrapper>
+      ) : null}
+    </>
+  );
+}
+
+const StyledCard = styled.div`
+  display: flex;
+  align-items: center;
+  border-top: 1px solid lightgray;
+  background-color: var(--background-color);
+  padding: 0.8rem;
+`;
+const MainInfoWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  flex-grow: 1;
+  gap: 0.6rem;
+  padding-left: 0.6rem;
+`;
+
+const MainInfoWrapperRow = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+`;
+
+const ExtraInfoWrapper = styled.div`
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
```

Comment:
are you sure, that this lines take effect, if it's not a `display: flex`?
i think you don't need it